server/ntb/scanpix/scanpix_datalayer.py

```
@@ -0,0 +1,365 @@
+                    created = criterion['range']['firstcreated']
+                    if 'gte' in created:
+                        start = created['gte'][0:10]
+                    if 'lte' in created:
+                        end = created['lte'][0:10]
+
+                # if there is a special start and no end it's one of the date buttons
+                if start and not end:
+                    if start == 'now-24H':
+                        data['timeLimit'] = 'last24 '
+                    if start == 'now-1w':
+                        data['timeLimit'] = 'lastweek '
+                    if start == 'now-1M':
+                        data['timeLimit'] = 'lastmonth '
+                elif start or end:
+                    if start:
+                        data['archived']['min'] = extract_date(start)
+                    if end:
+                        data['archived']['max'] = extract_date(end)
+
+            if 'terms' in criterion:
+                if 'type' in criterion.get('terms', {}):
+                    type_ = criterion['terms']['type']
+                    if type_ == CONTENT_TYPE.VIDEO:
+                        data['mainGroup'] = 'video'
+
+        offset, limit = int(req.get('from', '0')), max(10, int(req.get('size', '25')))
+        data['offset'] = offset
+        data['showNumResults'] = limit
+        r = self._request(url, data)
+        hits = self._parse_hits(r.json())
+        return ElasticCursor(docs=hits['docs'], hits={'hits': hits})
+
+    def _request(self, url, data):
+        """Perform GET request to given url.
+
+        It adds predefined headers and auth token if available.
+
+        :param url
+        :param data
+        """
+        r = requests.post(url, data=json.dumps(data), headers=self._headers, auth=(self._user, self._password))
+
+        if r.status_code < 200 or r.status_code >= 300:
+            logger.error('error fetching url=%s status=%s content=%s' % (url, r.status_code, r.content or ''))
```

maybe it should rather raise an error, because it continues to parse results even if there is an error